main.cpp

```diff
@@ -183,105 +302,264 @@ void Vinyl::moveToneArm(bool is12Inch)
+}
+
+void Envelope::loopEnvelope(bool loop)
+{
+    if (loop)
+    {
+        std::cout << "Start looping envelope" << std::endl;
+    }
+    else
+    {
+        std::cout << "Stop looping envelope" << std::endl;
+    }
+}
+
+void Envelope::invertEnvelope(bool invert)
+{
+    if(invert)
+    {
+        output = -output;
+    }
+}
 
 struct MixerAndMidi
 {
-    double synthGain = 0.9;
-    double extGain = 0.0;
-    double feedback = 0.0;
-    int midiChannel = 15;
-    bool clockSyncExt = false;
+    MixerAndMidi();
+    double synthGain, previousGain, extGain, feedback;
+    int midiChannel;
+    bool clockSyncExt;
 
-    int midiCC(int midiChannel, int midiCCIn);
-    int midiNote(int midiChannel, int keyPressed, bool keyIsPressed);
-    double signalFeedback( double feedback );
+    void changeMidiChannel(int newMidiChannel);
+    void muteAudio(bool muteAduio);
 };
 
+MixerAndMidi::MixerAndMidi()
+{
+    synthGain = 0.9;
+    previousGain = 0;
+    extGain = 0.0;
+    feedback = 0.0;
+    midiChannel = 15;
+    clockSyncExt = false;  
+    std::cout << "Instance of 'MixerAndMidi' has been constructed" << std::endl;
+}
+
+void MixerAndMidi::changeMidiChannel(int newMidiChannel)
+{
+    midiChannel = newMidiChannel;
+    std::cout << "Midi Channel = " << midiChannel << std::endl;
+}
+
+void MixerAndMidi::muteAudio(bool muteAudio)
+{
+    double newGain = muteAudio ? 0 : previousGain;
+    previousGain = synthGain;
+    synthGain = newGain;
+    std::cout << "Audio gain = " << synthGain << std::endl; 
+}
+
 struct NoiseGen
 {
-    bool noiseColourPink = false;
-    double noiseGain = 0.0;
-    double sahRate = 8.0;
-    double sahControl = 0.0; 
-    int routingSAH = 0;
+    NoiseGen();
+    bool noiseColourPink;
+    double noiseGain, sahRate, sahControl; 
+    int routingSAH;
 
-    double sahControlPitch( double sahControl, int routingSAH = 0  );
-    double sahControlCutoff( double sahControl, int routingSAH = 1); 
-    double sahControlLFORate( double sahControl, int routingSAH = 2  );
+    void changeNoiseColour();
 };
 
-struct Synth
+NoiseGen::NoiseGen()
 {
-    Oscillator oscillator; 
-    LFO lfo;
-    Envelope envelope; 
-    MixerAndMidi MixerAndMidi;
-    NoiseGen noiseGen;
+    noiseColourPink = false;
+    noiseGain = 0.0;
+    sahRate = 8.0;
+    sahControl = 0.0; 
+    routingSAH = 0;
+    std::cout << "Instance of 'NoiseGen' has been constructed" << std::endl;
+}
+
+void NoiseGen::changeNoiseColour()
+{
+    noiseColourPink ? noiseColourPink = false : noiseColourPink = true;
+}
 
-    double bassTone();
-    double snareDrum();
-    int sendMidiNote();
+struct Synth
+{
+    Synth();
+
+	Oscillator oscillator;
+	LFO lfo;
+	Envelope myEnvelope{5, 200, 0.5, 1000};
+	MixerAndMidi MixerAndMidi;
+	NoiseGen noiseGen;
+	float firmwareVersion;
+        
+    void updateFirmware(float currentFirmwareVersion, float latestFirmwareVersion);
+    void displayEnvelopeParameters();
+    double scope();
```

Comment:
`scope` is not an action.  it is a name.
Name things what they are.
Name functions based on the action they perform
Variables represent things. 
functions represent actions.
- [ ] revise